src/node_zlib.cc

```diff
@@ -1286,6 +1286,35 @@ struct MakeClass {
   }
 };
 
+template <typename T>
+T CallOnSequence(v8::Isolate* isolate,
+                 Local<Value> value,
+                 std::function<T(const char* data, size_t size)> callback) {
```

Comment:
Nit, feel free to ignore, but I think we can just skip the overhead that comes with `std::function` here:

```suggestion
template <typename T, typename F>
T CallOnSequence(v8::Isolate* isolate,
                 Local<Value> value,
                 F callback) {
```